packages/mongodb-chatbot-eval/src/Pipeline.ts

```diff
@@ -0,0 +1,135 @@
+  report: PipelineReportFunc
+) => Promise<void>;
+
+interface RunPipelineParams {
+  /**
+    A function that returns an {@link EvalConfig}.
+    This can be the same function as what you export from your configuration file.
+   */
+  configConstructor: ConfigConstructor;
+
+  /**
+    A function that takes `generate`, `evaluate`, and `report` functions
+    based on your configuration and runs a pipeline of commands.
+
+    These functions have the same behavior as the CLI commands,
+    creating the relevant data in their respective stores
+    and adding the command metadata.
+   */
+  pipelineFunc: PipelineFunc;
+}
+
+/**
+  Runs a pipeline of commands from configuration file.
+  This is a useful utility for chaining a group of commands.
+
+  For example, you may want to generate one set of converations,
+  then generate N evaluations for each conversation,
+  and then M reports for each evaluation.
+ */
+export async function runPipeline({
+  configConstructor,
+  pipelineFunc,
+}: RunPipelineParams): Promise<void> {
+  logger.info("Starting pipeline.");
+  logger.info("Constructing pipeline config.");
+
+  const config = await configConstructor();
+  const {
+    metadataStore,
+    generatedDataStore,
+    evaluationStore,
+    reportStore,
+    commands,
+    afterAll,
+  } = config;
+
+  const generateFunc: PipelineGenerateFunc = async (name) => {
+    assert(
+      commands.generate?.[name],
+      `Cannot find generate command with the name '${name}'.`
+    );
+    const { testCases, generator } = commands.generate[name];
+    const { metadata } = await generateDataAndMetadata({
+      testCases,
+      name,
+      generator,
+      generatedDataStore: generatedDataStore,
+      metadataStore: metadataStore,
+    });
+    return metadata;
+  };
+
+  const evaluateFunc: PipelineEvaluateFunc = async (
+    name,
+    generatedDataRunId
+  ) => {
+    assert(
+      commands.evaluate?.[name],
+      `Cannot find evaluate command with the name '${name}'.`
+    );
+    const { evaluator } = commands.evaluate[name];
+    const { metadata } = await generateEvalsAndMetadata({
+      name,
+      evaluator,
+      generatedDataRunId,
+      generatedDataStore: generatedDataStore,
+      evaluationStore: evaluationStore,
+      metadataStore: metadataStore,
+    });
+    return metadata;
+  };
+
+  const reportFunc: PipelineReportFunc = async (name, evaluationRunId) => {
+    assert(
+      commands.report?.[name],
+      `Cannot find report command with the name '${name}'.`
+    );
+    const { reporter } = commands.report[name];
+    const { metadata } = await generateReportAndMetadata({
+      name,
+      reportEvalFunc: reporter,
+      reportStore: reportStore,
+      evaluationStore: evaluationStore,
+      metadataStore: metadataStore,
+      evaluationRunId,
+    });
+    return metadata;
+  };
+
+  try {
+    logger.info("Running pipeline actions.");
+    await pipelineFunc(generateFunc, evaluateFunc, reportFunc);
```

Comment:
```suggestion
    await pipeline(generateFunc, evaluateFunc, reportFunc);
```

Didn't change the other `fooFunc` names because they're internal. Up to you if you'd want to change them too for consistency.